app/models/git_repository.rb

```diff
@@ -39,6 +39,20 @@ def commit_from_ref(git_reference)
     description.split('-').last.sub(/^g/, '')
   end
 
+  def tag_from_ref(git_reference)
+    Dir.chdir(repo_cache_dir) do
+      tag = IO.popen(['git', 'describe', '--tags', git_reference], err: [:child, :out]) do |io|
```

Comment:
The test output covering the nil case gets an error dumped to stderr that I was trying to avoid.